This patch file is from Qualcomm Technologies, Inc.
and is provided under the following license:

Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
SPDX-License-Identifier: BSD-3-Clause-Clear

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Susheel Nyamala <snyamala@qti.qualcomm.com>
Date: Wed, 04 Jun 2026 00:00:00 +0000
Subject: [PATCH] Fix null issue for RfcJsonInit null filename

CosaDmlDiRfcDefaultsInit calls RfcJsonInit with NULL as jsonFileName.
GCC inlines RfcJsonInit and raises -Werror=nonnull because access()
is declared __nonnull((1)). Add an explicit NULL guard before the
access() call to make the NULL path safe and silence the error.

Signed-off-by: Susheel Nyamala <snyamala@qti.qualcomm.com>
---
 source-arm/TR-181/board_sbapi/cosa_deviceinfo_apis.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/source-arm/TR-181/board_sbapi/cosa_deviceinfo_apis.c b/source-arm/TR-181/board_sbapi/cosa_deviceinfo_apis.c
--- a/source-arm/TR-181/board_sbapi/cosa_deviceinfo_apis.c
+++ b/source-arm/TR-181/board_sbapi/cosa_deviceinfo_apis.c
@@ -5013,6 +5013,11 @@ RfcJsonInit
     char* data = NULL;
     int len ;
 
+    if (jsonFileName == NULL)
+    {
+        return ANSC_STATUS_FAILURE;
+    }
+
     if (access(jsonFileName, F_OK) != 0)
     {
         return ANSC_STATUS_FAILURE;
-- 
2.34.1

